Source/Particles/Collision/CollisionBase.H

```diff
@@ -0,0 +1,31 @@
+/* Copyright 2020 David Grote
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_PARTICLES_COLLISION_COLLISIONBASE_H_
+#define WARPX_PARTICLES_COLLISION_COLLISIONBASE_H_
+
+#include <AMReX_ParmParse.H>
+#include <AMReX_Vector.H>
+
+// Forward declaration needed
+class MultiParticleContainer;
+
+class CollisionBase
+{
+public:
+
+    CollisionBase (std::string collision_name);
+
+    virtual void doCollisions (amrex::Real /*cur_time*/, MultiParticleContainer* /*mypc*/ ){}
+
```

Comment:
We could potentially remove the copy and move constructor and the assignment operator of this class to avoid potential object slicing fallpits:
https://www.geeksforgeeks.org/object-slicing-in-c/

We also need a virtual destructor here: https://www.geeksforgeeks.org/virtual-destructor/
```suggestion

    CollisionBase(CollisionBase const &) = delete;
    CollisionBase(CollisionBase &&) = delete;
    CollisionBase & operator=(CollisionBase const &) = delete;
    
    virtual ~CollisionBase() = default;
```